projects/core/src/cms/services/dynamic-attribute.service.ts

```diff
@@ -1,18 +1,35 @@
-import { Injectable, Renderer2 } from '@angular/core';
+import { Inject, Injectable, Renderer2 } from '@angular/core';
 import { SmartEditService } from '../../smart-edit/services/smart-edit.service';
+import { ComponentDecorator } from '../decorators/component-decorator';
+import { PageDecorator } from '../decorators/page-decorator';
+import { SlotDecorator } from '../decorators/slot-decorator';
 import { ContentSlotComponentData } from '../model/content-slot-component-data.model';
 import { ContentSlotData } from '../model/content-slot-data.model';
+import { Page } from '../model/page.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class DynamicAttributeService {
-  constructor(protected smartEditService: SmartEditService) {}
+  /**
+   * @deprecated since 3.0
+   */
+  constructor(smartEditService?: SmartEditService);
+  constructor(
+    // TODO: remove this SmartEditService in major release
+    protected smartEditService?: SmartEditService,
+    @Inject(ComponentDecorator)
```

Comment:
Shouldn't we use @Optional here also?

Reply:
Also, to make it work with lazy loading, we will have to switch from Inject to UnifiedInjector thingy

Reply:
I will switch to UnifedInjector in that smartedit lib ticket.

Reply:
👍